File: 009/solution_009.py
def find_triplets(target_sum):
    for a in xrange(target_sum):
        for b in xrange(target_sum):
            for c in xrange(target_sum):
                if (a < b < c) \
                and (a * a + b * b == c * c) \
                and (a + b + c == target_sum):
                    return (a, b, c)


def solution(target):
    triplets = find_triplets(target)
    product = triplets[0] * triplets[1] * triplets[2]
    return product


if __name__ == '__main__':
    print solution(1000)

File: 009/solution_009.erl
-module(solution_009).
-export([start/0]).


triplets(Target) ->
    Candidates = lists:seq(1, Target),
    [
        {A, B, C} ||
        A <- Candidates, B <- Candidates, C <- Candidates,
        (A < B) and (B < C),
        A + B + C =:= Target,
        (A * A) + (B * B) =:= C * C
    ].


solution(Target) ->
    Triplets = triplets(Target),
    Products = [A * B * C || {A, B, C} <- Triplets],
    Products.


start() ->
    io:format("~p~n", [solution(1000)]).
